### PR TITLE
Apply and enforce format rules

### PR DESCRIPTION
**Pull request by Sgitario**
Checkstyle plugin is used to enforce the format rules and hence the build will fail if a format rule is violated. 
The rules are specified in the `checkstyle.xml` file and mimic the ones defined in Quarkus upstream.
As part of this PR, I didn't format the ".properties" files or ".html" files.
Some rules:
- Avoid magic values/numbers
- Method name must match a reg expression
- Field declaration order
- Unused imports
- Imports in right order
- XML whitespaces vs tabs

### CODE REVIEW

**Comment by Sgitario** [comment]
In Draft to verify tests are passing.

**Comment by Sgitario** [comment]
Fixed failures and ready to be reviewed now.

**Comment by Sgitario** [comment]
Rebased using the latest PR merged (module 601).

File: 603-spring-web-smallrye-openapi/src/main/java/io/quarkus/qe/books/persistence/model/Book.java [hunk 1]
```diff
@@ -55,35 +57,40 @@ public void setAuthor(String author) {
...
-        int result = 1;
-        result = prime * result + ((author == null) ? 0 : author.hashCode());
-        result = prime * result + (int) (id ^ (id >>> 32));
-        result = prime * result + ((title == null) ? 0 : title.hashCode());
-        return result;
+        return Objects.hash(author, id, title);
     }
 
     @Override
     public boolean equals(Object obj) {
```
Comment (jsmrcka) [inline 603-spring-web-smallrye-openapi/src/main/java/io/quarkus/qe/books/persistence/model/Book.java:64]: I know it is beyond the scope of this PR, but I would consider to use some 3rd-party help here:
```
        org.apache.commons.lang3.builder.EqualsBuilder.reflectionEquals(this, obj);
```
WDYT?

File: 603-spring-web-smallrye-openapi/src/test/java/org/acme/spring/web/OpenApiTest.java [hunk 2]
```diff
@@ -8,40 +8,34 @@
...
 import io.restassured.response.Response;
 
 @QuarkusTest
-@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 public class OpenApiTest {
-    private static Response response;
+    private Response response;
 
-    private void callOpenApiEndpoint() {
+    @BeforeEach
```
Comment (jsmrcka) [inline 603-spring-web-smallrye-openapi/src/test/java/org/acme/spring/web/OpenApiTest.java:23]: Thanks for sorting this out. The original intention was to save a few requests, but the independence of the tests is better. ;)

File: pom.xml [hunk 3]
```diff
@@ -1,207 +1,240 @@
...
+                        <version>${surefire-plugin.version}</version>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>integration-test</goal>
+                                    <goal>verify</goal>
+                                </goals>
+                                <configuration>
+                                    <systemProperties>
+                                        <native.image.path>${project.build.directory}/${project.build.finalName}-runner
```
Comment (jsmrcka) [inline pom.xml:224]: I don't particularly fancy the looks of this:
```
<tag>too-long-cdata
</tag>
```
and would prefer something like:
```
<tag>
    too-long-cdata
</tag>
```
But I'm not sure how hard it is to automate it.

File: 004-quarkus-HHH-and-HV/src/main/java/io/quarkus/qe/hibernate/validator/SomeEntity.java [hunk 4]
```diff
@@ -43,11 +46,11 @@ public void setId(Long id) {
...
 
-    public Set<SomeChildEntity> getSomeChilds() {
-        return someChilds;
+    public Set<SomeChildEntity> getSomeChildren() {
+        return someChildren;
     }
 
-    public void setSomeChilds(Set<SomeChildEntity> someChilds) {
-        this.someChilds = someChilds;
+    public void setSomeChildren(Set<SomeChildEntity> someChilds) {
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 004-quarkus-HHH-and-HV/src/main/java/io/quarkus/qe/hibernate/validator/SomeEntity.java:53]: Thanks for spotting! Fixed.

File: 301-quarkus-vertx-kafka/src/test/java/io/quarkus/qe/kafka/KafkaCommonTest.java [hunk 5]
```diff
@@ -25,11 +24,15 @@
...
 
+import io.restassured.response.Response;
+
 abstract class KafkaCommonTest {
-    private final static String jaegerEndpoint = "http://localhost:16686/api/traces";
-    static final String NATIVE = "native";
-    static final String QUARKUS_PROFILE = "quarkus.profile";
-    static final boolean IS_NATIVE = System.getProperty(QUARKUS_PROFILE, "").equals(NATIVE);
+
+    protected static final String NATIVE = "native";
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 301-quarkus-vertx-kafka/src/test/java/io/quarkus/qe/kafka/KafkaCommonTest.java:31]: Done

File: 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/vertx/web/AbstractCommonTest.java [hunk 6]
```diff
@@ -98,6 +109,7 @@ protected String JWT(Invalidity invalidity, String... groups) {
                 jwt = JWTAuth.create(vertx.getDelegate(), new JWTAuthOptions()
                         .addPubSecKey(getPubSecKeyOptions(authConfig)));
                 break;
+            default:
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 302-quarkus-vertx-jwt/src/test/java/io/quarkus/qe/vertx/web/AbstractCommonTest.java:112]: Good suggestion! Changed.

File: 603-spring-web-smallrye-openapi/src/main/java/io/quarkus/qe/books/persistence/model/Book.java (hunk 1 above)
Comment (Sgitario) [reply 603-spring-web-smallrye-openapi/src/main/java/io/quarkus/qe/books/persistence/model/Book.java:64]: I also think the equals methods should be refactored... but as you said, not of scope of this PR. Feel free to create a card in Trello and propose it, or maybe also something to discuss in our next retrospective.

File: pom.xml (hunk 3 above)
Comment (Sgitario) [reply pom.xml:224]: +1
changed

File: 004-quarkus-HHH-and-HV/src/main/java/io/quarkus/qe/hibernate/validator/SomeEntity.java [hunk 7]
```diff
@@ -43,11 +46,11 @@ public void setId(Long id) {
...
+        this.someChildren = someChilds;
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 004-quarkus-HHH-and-HV/src/main/java/io/quarkus/qe/hibernate/validator/SomeEntity.java:54]: Fixed.

File: 012-quarkus-kafka-streams/src/main/java/io/quarkus/qe/kafka/producer/LoginEventsProducer.java [hunk 8]
```diff
@@ -1,52 +1,56 @@
...
+
+import io.quarkus.qe.kafka.model.LoginAttempt;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.reactive.messaging.kafka.Record;
+import io.vertx.core.json.Json;
+
 @ApplicationScoped
 public class LoginEventsProducer {
 
+    private static final int SEND_RECORD_EVERY_MS = 100;
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 012-quarkus-kafka-streams/src/main/java/io/quarkus/qe/kafka/producer/LoginEventsProducer.java:22]: I prefer keeping it as SEND_RECORD_EVERY_MS...

File: 013-quarkus-oidc-restclient/src/main/java/io/quarkus/qe/model/Score.java [hunk 9]
```diff
@@ -32,9 +33,6 @@ public void setTeamB(int teamB) {
 
     @Override
     public String toString() {
-        return "Score{" +
-                "teamA=" + teamA +
-                ", teamB=" + teamB +
-                '}';
+        return "Score{" + "teamA=" + teamA + ", teamB=" + teamB + '}';
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 013-quarkus-oidc-restclient/src/main/java/io/quarkus/qe/model/Score.java:36]: What an eye! 
Changed.

File: 304-quarkus-vertx-routes/src/main/java/io/quarkus/qe/validation/ValidationOnRequestParamsRouteHandler.java [hunk 10]
```diff
@@ -11,15 +11,18 @@
 
 @RouteBase(path = "/validate")
 public class ValidationOnRequestParamsRouteHandler {
+
+    private static final int SIZE = 3;
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 304-quarkus-vertx-routes/src/main/java/io/quarkus/qe/validation/ValidationOnRequestParamsRouteHandler.java:15]: Renaming it to `FIRST_PARAM_SIZE` makes the line lenght to be longer than 128 and this is not allowed.
I don't think this is too important to be renamed so far.

File: 304-quarkus-vertx-routes/src/main/java/io/quarkus/qe/validation/ValidationOnRequestParamsRouteHandler.java [hunk 11]
```diff
@@ -11,15 +11,18 @@
...
+
     @Route(methods = HttpMethod.GET, path = "/request-single-param/:firstParam")
     boolean validateRequestSingleParam(
-            @Param("firstParam") @Size(min = 3, max = 3, message = "Param must have 3 characters") String param) {
+            @Param("firstParam") @Size(min = SIZE, max = SIZE, message = "Param must have 3 characters") String param) {
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 304-quarkus-vertx-routes/src/main/java/io/quarkus/qe/validation/ValidationOnRequestParamsRouteHandler.java:19]: Same as above.

File: 303-quarkus-vertx-sql/src/main/java/io/quarkus/qe/vertx/sql/handlers/FlightsHandler.java [hunk 12]
```diff
@@ -58,16 +60,20 @@ void search(@Body @Valid QueryFlightSearch query, RoutingContext context) {
...
+        Flight.findAllAsList(connection)
+                .onFailure().invoke(context::fail)
+                .subscribe().with(flights -> context.response().end(Record.toJsonStringify(flights)));
     }
 
     @Operation(summary = "Retrieve all flights by origin and destination")
-    @APIResponse(responseCode = "200", content = @Content(mediaType = "application/json", schema = @Schema(type = SchemaType.ARRAY, implementation = Flight.class)))
+    @APIResponse(responseCode = "200",
+            content = @Content(mediaType = "application/json",
+                    schema = @Schema(type = SchemaType.ARRAY, implementation = Flight.class)))
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 303-quarkus-vertx-sql/src/main/java/io/quarkus/qe/vertx/sql/handlers/FlightsHandler.java:76]: This is automatically done by my IDE and the formatter is not complaining about it.

File: 303-quarkus-vertx-sql/src/main/java/io/quarkus/qe/vertx/sql/handlers/FlightsHandler.java [hunk 13]
```diff
@@ -58,16 +60,20 @@ void search(@Body @Valid QueryFlightSearch query, RoutingContext context) {
     }
 
     @Operation(summary = "Retrieve all flights")
-    @APIResponse(responseCode = "200", content = @Content(mediaType = "application/json", schema = @Schema(type = SchemaType.ARRAY, implementation = Flight.class)))
+    @APIResponse(responseCode = "200",
+            content = @Content(mediaType = "application/json",
+                    schema = @Schema(type = SchemaType.ARRAY, implementation = Flight.class)))
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply 303-quarkus-vertx-sql/src/main/java/io/quarkus/qe/vertx/sql/handlers/FlightsHandler.java:65]: This is automatically done by my IDE and the formatter is not complaining about it.

**Comment by Sgitario** [comment]
I think I've already addressed/replied all your requested changes.
I've also formatted the module 009 that I missed (thanks for spotting @kshpak )
Would you mind to have another look?

File: 300-quarkus-vertx-webClient/src/test/java/io/quarkus/qe/vertx/webclient/ChuckNorrisResourceTest.java [hunk 14]
```diff
@@ -29,17 +14,35 @@
...
 public class ChuckNorrisResourceTest {
-    private final static String jaegerEndpoint = "http://localhost:16686/api/traces";
-    static final String EXPECTED_ID = "aBanNLDwR-SAz7iMHuCiyw";
-    static final String EXPECTED_VALUE = "Chuck Norris has already been to mars; that why there's no signs of life";
-    static final int DELAY = 3500; // must be greater than vertx.webclient.timeout-sec
-    static final String QUARKUS_PROFILE = "quarkus.profile";
-    static final String NATIVE = "native";
-    static final boolean IS_NATIVE = System.getProperty(QUARKUS_PROFILE, "").equals(NATIVE);
+    private static final String JAEGER_ENDPOINT = "http://localhost:16686/api/traces";
+
```
Comment (kshpak) [inline 300-quarkus-vertx-webClient/src/test/java/io/quarkus/qe/vertx/webclient/ChuckNorrisResourceTest.java:39]: is this blank line really needed?

File: 301-quarkus-vertx-kafka/pom.xml [hunk 15]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0"?>
```
Comment (kshpak) [inline 301-quarkus-vertx-kafka/pom.xml:1]: pls change repository URL below to HTTPS; as maven 3.8.1 doesn't recognize insecure connections.

**Review by kshpak — CHANGES_REQUESTED** [review]
can HTTPS for repositories be added as a rule?
`<repository>
            <id>confluent</id>
            <url>http://packages.confluent.io/maven/</url>
            <snapshots>
                <enabled>false</enabled>
            </snapshots>
        </repository>`

File: 300-quarkus-vertx-webClient/src/test/java/io/quarkus/qe/vertx/webclient/ChuckNorrisResourceTest.java (hunk 14 above)
Comment (Sgitario) [reply 300-quarkus-vertx-webClient/src/test/java/io/quarkus/qe/vertx/webclient/ChuckNorrisResourceTest.java:39]: Removed

File: 301-quarkus-vertx-kafka/pom.xml (hunk 15 above)
Comment (Sgitario) [reply 301-quarkus-vertx-kafka/pom.xml:1]: Out of scope of this PR. Please create a separated PR to address this.

**Comment by Sgitario** [comment]
The failures are unrelated to this PR and will be addressed as part of another PR.

**Review by kshpak — APPROVED** [review]

**Review by jsmrcka — APPROVED** [review]

**Pull request merged**